src/main/java/mops/businesslogic/FileInfoServiceImpl.java

```diff
@@ -0,0 +1,67 @@
+package mops.businesslogic;
+
+import lombok.AllArgsConstructor;
+import mops.exception.MopsException;
+import mops.persistence.FileInfoRepository;
+import mops.persistence.file.FileInfo;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+@AllArgsConstructor
+public class FileInfoServiceImpl implements FileInfoService {
+
+    /**
+     * reference to fileInfoRepository.
+     */
+    private final FileInfoRepository fileInfoRepo;
+
+    /**
+     * @param dirId directory id
+     * @return a list of files in that directory
+     */
+    @Override
+    public List<FileInfo> fetchAllFilesInDirectory(long dirId) throws MopsException {
+        return fileInfoRepo.getAllFileInfoByDirectory(dirId);
+    }
+
+    /**
+     * @param fileId file id
+     * @return a FileInfo object
+     */
+    @SuppressWarnings("PMD")
+    @Override
+    public FileInfo fetchFileInfo(long fileId) throws MopsException {
+        return fileInfoRepo.findById(fileId).orElseThrow(() -> new MopsException("..."));
```

Comment:
Shouldn't we use a message and an explicit sub type of `MopsException`?

Reply:
Yes, but nobody cared for exceptions so far, so we don't have any explicit sub types. I believe ExceptionHandling is a whole different thing we still need to do. To do it in this PR would be way to much.

Reply:
Look at the other Services. There are a lot of types of custom exceptions. Just write a nice message.

Reply:
> Yes, but nobody cared for exceptions so far, so we don't have any explicit sub types. I believe ExceptionHandling is a whole different thing we still need to do. To do it in this PR would be way to much.

You are absolutely right about the Exception Handling, but this is not handling but throwing. You just throw a custom exception and do not care about the handling. This should be only a couple of lines.

It it also part of the service's logic what exception is thrown.